src/webworker.js

```diff
@@ -1,17 +1,19 @@
 importScripts('./pyodide.js')
 
 onmessage = async function(e) {
-  await loadPyodide({indexURL : '{{ PYODIDE_BASE_URL }}'});
-  const data = e.data;
-  for (let key of Object.keys(data)) {
-    if (key !== 'python') {
-      // Keys other than python must be arguments for the python script.
-      // Set them on self, so that `from js import key` works.
-      self[key] = data[key];
+  try {
+    const data = e.data;
+    for (let key of Object.keys(data)) {
+      if (key !== 'python') {
+        // Keys other than python must be arguments for the python script.
+        // Set them on self, so that `from js import key` works.
+        self[key] = data[key];
+      }
     }
-  }
 
-  try {
+    if (typeof self.pyodide !== "undefined") {
+      await loadPyodide({packageIndexURL : '{{ PYODIDE_BASE_URL }}'});
+    }
     self.postMessage(
         {results : await self.pyodide.runPythonAsync(data.python)});
```

Comment:
This looks weird to me. If `self.pyodide === "undefined"` then you skip `loadPyodide` but then `self.pyodide.runPythonAsync` is definitely going to fail too. Maybe we should just throw an error if it's undefined? When would we expect to run into that condition?

Reply:
Indeed, that was meant to be a `===` not `!==`.

The idea was to initialize pyodide on first recieved message, and avoid initializing it on the following messages, in order to move in the direction of your other comment in https://github.com/pyodide/pyodide/pull/1404#discussion_r605753753 

The fact that it didn't fail for me locally, means I was likely not testing this actual PR. I'll try again, maybe I'll now be able to reproduce other failures in this PR.

Reply:
Oh right, that makes a lot more sense!